controllers/nstemplateset/features_test.go

```diff
@@ -0,0 +1,119 @@
+			objFeature:            p(""),
+			nsTemplateSetFeatures: p(""),
+			expectedToBeCreated:   false,
+		},
+		{
+			name:                  "object with no feature annotation should be created when NSTemplateSet has an empty feature annotation",
+			objFeature:            nil,
+			nsTemplateSetFeatures: p(""),
+			expectedToBeCreated:   true,
+		},
+		{
+			name:                  "object with an empty feature annotation should not be created when NSTemplateSet has some features enabled",
+			objFeature:            p(""),
+			nsTemplateSetFeatures: p("feature-1"),
+			expectedToBeCreated:   false,
+		},
+		{
+			name:                  "object with a feature should not be created when NSTemplateSet has an empty feature annotation",
+			objFeature:            p("feature-1"),
+			nsTemplateSetFeatures: p(""),
+			expectedToBeCreated:   false,
+		},
+		{
+			name:                  "object with no feature annotation should be created when no features are enabled in NSTemplateSet",
+			objFeature:            nil,
+			nsTemplateSetFeatures: nil,
+			expectedToBeCreated:   true,
+		},
+		{
+			name:                  "object with no feature annotation should be created when some features are enabled in NSTemplateSet",
+			objFeature:            nil,
+			nsTemplateSetFeatures: p("feature-1,feature-2"),
+			expectedToBeCreated:   true,
+		},
+		{
+			name:                  "object with a feature annotation should not be created when no features are enabled in NSTemplateSet",
+			objFeature:            p("feature-1"),
+			nsTemplateSetFeatures: nil,
+			expectedToBeCreated:   false,
+		},
+		{
+			name:                  "object with a feature annotation should not be created when different features are enabled in NSTemplateSet",
+			objFeature:            p("feature-1"),
+			nsTemplateSetFeatures: p("feature-2"),
+			expectedToBeCreated:   false,
+		},
+		{
+			name:                  "object with a feature which is among enabled features should be created",
+			objFeature:            p("feature-2"),
+			nsTemplateSetFeatures: p("feature-1,feature-2,feature-3"),
+			expectedToBeCreated:   true,
+		},
+	}
+	for _, testRun := range tests {
+		t.Run(testRun.name, func(t *testing.T) {
+			// given
+			obj := objectWithFeature(testRun.objFeature)
+			nsTmplSet := newNSTmplSet("dummy-namespace", "dummy-name", "dummy-tier")
+			if testRun.nsTemplateSetFeatures != nil {
+				nsTmplSet.Annotations = map[string]string{
+					toolchainv1alpha1.FeatureToggleNameAnnotationKey: *testRun.nsTemplateSetFeatures,
+				}
+			}
+
+			// when
+			should := shouldCreate(obj, nsTmplSet)
+
+			// then
+			assert.Equal(t, testRun.expectedToBeCreated, should)
+		})
+	}
+}
+
+func TestReallySplit(t *testing.T) {
+	assert.Empty(t, splitCommaSeparatedList(""))
+	assert.Equal(t, strings.Split("1,2,3", ","), splitCommaSeparatedList("1,2,3"))
+	assert.Equal(t, strings.Split("1", ","), splitCommaSeparatedList("1"))
+}
+
+func p(s string) *string {
+	return &s
+}
```

Comment:
you could use a generic version of this:
```suggestion
func p[T any](v T) *T {
	return &v
}
```
export it and move it somewhere to make it re-usable

Reply:
I was thinking about moving it to common but it's just one-line function... Not sure if it's worth it?

Reply:
Maybe later, it's fine. I just wanted to let you know about the generic version as well.

TBH, I would expect such a function to be provided by Go library itself.